app/utils/detection.js

```diff
@@ -0,0 +1,131 @@
+import UAParser from 'ua-parser-js';
+import semverCompare from 'semver-compare';
+import identity from 'lodash/identity';
+
+const DEFAULT_SUPPORTED_BROWSERS_CONFIG = {
+  desktop: [{
+    browser: 'firefox', minversion: 41,
+  }, {
+    browser: 'ie', versions: [11, 'edge'],
+  }, {
+    browser: 'chrome', minversion: 45,
+  }, {
+    browser: 'edge',
+  }, {
+    os: 'mac os', minos: '10.10.0', browser: 'safari', minversion: 8,
+  }],
+  tablet: [{
+    os: 'ios', minos: '9', browser: 'mobile safari',
+  }, {
+    os: 'android', minos: '5.0', browser: 'chrome',
+  }, {
+    browser: 'ie', versions: [11, 'edge'],
+  }, {
+    browser: 'edge',
+  }],
+  mobile: [{
+    os: 'ios', minos: '9', browser: 'mobile safari',
+  }, {
+    os: 'ios', minos: '5.0', browser: 'chrome',
+  }, {
+    os: 'android', minos: '5.0', browser: 'chrome', minversion: 50,
+  }],
+};
+
+export default class Detection {
+  parser = new UAParser();
+
+  constructor(config = DEFAULT_SUPPORTED_BROWSERS_CONFIG, isInAppBrowserSupported = true) {
+    this.supportedBrowsersConfig = config;
+    this.isInAppBrowserSupported = isInAppBrowserSupported;
+  }
+
+  get isInAppBrowser() {
+    return (
+      (this.ua.indexOf('FBAN') > -1) ||
+      (this.ua.indexOf('FBAV') > -1) ||
+      (this.ua.indexOf('Twitter') > -1)
+    );
+  }
+
+  get device() {
+    return this.parser.getDevice();
+  }
+
+  get ua() {
+    return this.parser.getUA();
+  }
+
+  get browser() {
+    return this.parser.getBrowser();
+  }
+
+  get os() {
+    return this.parser.getOS();
+  }
+
+  get deviceType() {
+    const { type = 'desktop' } = this.device;
+    return type;
+  }
+
+  compareVersions(a, b) {
+    if (typeof a === 'string' || a instanceof String) {
+      return semverCompare(a, b) <= 0;
+    }
+
+    return a <= parseInt(b, 10);
+  }
+
+  isSupported() {
+    document.documentElement.className += ` device-${this.deviceType}`;
+
+    if (this.isInAppBrowser) {
+      document.documentElement.className += ' in-app-browser';
+      return this.isInAppBrowserSupported;
+    }
+
+    const { version: browserVersion } = this.browser;
+
+    const isSupported = !this.supportedBrowsersConfig[this.deviceType].every(options => {
+      const checkedOptions = Object.keys(options).map(optionKey => {
+        const value = options[optionKey];
+
+        switch (optionKey) {
```

Comment:
Can you use an object instead of switch case?

Switch is less readable and "mistake prone".